src/track/keyfactory.cpp

```diff
@@ -22,27 +22,34 @@ Keys KeyFactory::loadKeysFromByteArray(const QString& keysVersion,
 }
 
 // static
-Keys KeyFactory::makeBasicKeys(mixxx::track::io::key::ChromaticKey global_key,
-                               mixxx::track::io::key::Source source) {
+Keys KeyFactory::makeBasicKeys(
+        mixxx::track::io::key::ChromaticKey global_key,
+        mixxx::track::io::key::Source source) {
     KeyMap key_map;
     key_map.set_global_key(global_key);
+    QString global_key_text = KeyUtils::keyToString(global_key, KeyUtils::KeyNotation::Custom);
+    key_map.set_global_key_text(global_key_text.toStdString());
```

Comment:
this will cause a bunch of more mutex locks in a function that seems cheap. Are you sure this won't cause any problems?

Reply:
Which mutex do you mean? When allocating memory? 
Do you have an alternative version in mind?

Reply:
`KeyUtils::keyToString` locks a mutex when using `KeyNotation::Custom`.

Reply:
> KeyUtils::keyToString locks a mutex when using KeyNotation::Custom.

Ah, I see. This has only a limited scope and protects only against concurrent editing in the preferences. 
We may consider to use a copy during editing and atomic swap to get around it. 

Here is helps to have always a consistent key_text which will become the leading source of date with the next PR.